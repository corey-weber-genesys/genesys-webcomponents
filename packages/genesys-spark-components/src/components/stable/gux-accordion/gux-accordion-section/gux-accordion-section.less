@import (reference) '../../../../style/color.less';
@import (reference) '../../../../style/focus.less';
@import (reference) '../../../../style/spacing.less';
@import (reference) '../../../../style/typography.less';

:host {
  color: @gux-black-50;
  -webkit-user-select: none;
  user-select: none;
  background-color: @gux-grey-100;

  &:first-child {
    .gux-header {
      border-top: 1px solid @gux-grey-60;
    }
  }
}

section.gux-disabled {
  cursor: default;
  opacity: 0.5;

  > * {
    pointer-events: none;
  }
}

.gux-header {
  all: unset;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: center;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  padding: @gux-spacing-small @gux-spacing-small @gux-spacing-small
    @gux-spacing-medium;
  margin: 0;
  cursor: pointer;
  border-bottom: 1px solid @gux-grey-60;

  .gux-header-text {
    flex: 1 1 auto;
    align-self: auto;
    padding-right: @gux-spacing-xs;
    text-align: left;
  }

  &.gux-reverse-headings {
    & .gux-header-text {
      display: flex;
      flex-direction: column-reverse;
    }
  }

  ::slotted([slot='header']) {
    padding: 0;
    margin: 0;
    font-size: 12px !important;
    font-weight: bold !important;
    line-height: @gux-spacing-medium !important;
  }

  ::slotted([slot='subheader']) {
    padding: 0;
    margin: @gux-spacing-3xs 0 0;
    font-size: 11px !important;
    font-weight: 400 !important;
    color: @gux-black-80;

    .gux-reverse-headings& {
      margin: 0 0 @gux-spacing-3xs 0;
    }
  }

  ::slotted([slot='icon']) {
    flex: 0 0 auto;
    align-self: flex-start;
    width: @gux-spacing-medium;
    height: @gux-spacing-medium;
    margin-right: @gux-spacing-xs;
    color: @gux-black-100;

    .gux-reverse-headings& {
      align-self: flex-end;
    }
  }

  .gux-header-icon {
    flex: 0 1 auto;
    align-items: center;
    align-self: auto;
    margin-left: @gux-spacing-xs;
    color: @gux-black-100;
    transition: transform 0.5s ease;
    transform-origin: center;

    gux-icon {
      width: @gux-spacing-medium;
      height: @gux-spacing-medium;
    }

    &.gux-arrow-position-left {
      order: -1;
      margin: 0 @gux-spacing-xs 0 0;
    }

    &.gux-expanded {
      transform: rotate(-180deg);
    }
  }

  &:focus-visible {
    .gux-focus-ring();

    .gux-header-icon {
      color: @gux-black-50;
    }
  }

  &:hover {
    .gux-header-icon {
      color: @gux-black-50;
    }
  }
}

.gux-content {
  box-sizing: border-box;
  display: none;
  border-bottom: 1px solid @gux-grey-60;

  &.gux-expanded {
    display: block;
  }

  &.gux-text-content-layout {
    padding: @gux-spacing-medium;
  }
}
