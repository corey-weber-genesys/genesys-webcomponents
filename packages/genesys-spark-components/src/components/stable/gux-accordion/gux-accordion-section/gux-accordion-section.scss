@import '../../../../style/color.scss';
@import '../../../..//style/focus.scss';
@import '../../../../style/spacing.scss';
@import '../../../../style/typography.scss';

:host {
  -webkit-user-select: none;
  user-select: none;

  &:first-child {
    .gux-header {
      border-top: var(--gse-ui-accordion-wrapper-dividerBorder-width)
        var(--gse-ui-accordion-wrapper-dividerBorder-style)
        var(--gse-ui-accordion-wrapper-dividerBorder-color);
    }
  }
}

section.gux-disabled {
  cursor: default;
  opacity: 0.5;

  > * {
    pointer-events: none;
  }
}

.gux-header {
  all: unset;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-content: center;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  padding: var(--gse-ui-accordion-header-padding);
  margin: 0;
  cursor: pointer;
  border-bottom: var(--gse-ui-accordion-wrapper-dividerBorder-width)
    var(--gse-ui-accordion-wrapper-dividerBorder-style)
    var(--gse-ui-accordion-wrapper-dividerBorder-color);

  .gux-header-text {
    flex: 1 1 auto;
    align-self: auto;
    padding-right: var(--gux-spacing-xs);
    text-align: left;
    color: var(--gse-ui-accordion-header-default-foreground-labelColor);
    font-weight: (--gse-ui-accordion-header-label-defaultText-fontWeight);
  }

  &.gux-reverse-headings {
    & .gux-header-text {
      display: flex;
      flex-direction: column-reverse;
    }
  }

  ::slotted([slot='header']) {
    padding: 0;
    margin: 0;
    font-size: var(
      --gse-ui-accordion-header-label-defaultText-fontFamily
    ) !important;
    font-weight: var(
      --gse-ui-accordion-header-label-defaultText-fontWeight
    ) !important;
    line-height: var(
      --gse-ui-accordion-header-label-defaultText-lineHeight
    ) !important;
  }

  ::slotted([slot='subheader']) {
    padding: 0;
    margin: var(--gux-spacing-3xs) 0 0;
    font-size: var(
      --gse-ui-accordion-contentItem-defaultText-fontSize
    ) !important;
    font-weight: var(
      --gse-ui-accordion-contentItem-defaultText-fontWeight
    ) !important;
    color: var(--gse-ui-accordion-contentItem-foregroundColor);

    .gux-reverse-headings {
      margin: 0 0 var(--gux-spacing-3xs) 0;
    }
  }

  ::slotted([slot='icon']) {
    flex: 0 0 auto;
    align-self: flex-start;
    width: var(--gse-ui-icon-size-sm);
    height: var(--gse-ui-icon-size-sm);
    margin-right: var(--gux-spacing-xs);
    color: var(--gse-ui-accordion-header-default-foreground-chevronIcon-closed);

    .gux-reverse-headings {
      align-self: flex-end;
    }
  }

  .gux-header-icon {
    flex: 0 1 auto;
    align-items: center;
    align-self: auto;
    margin-left: var(--gux-spacing-xs);
    color: var(--gse-ui-accordion-header-default-foreground-chevronIcon-closed);
    transition: transform 0.5s ease;
    transform-origin: center;

    gux-icon {
      width: var(--gse-ui-accordion-header-chevronIcon);
      height: var(--gse-ui-accordion-header-chevronIcon);
    }

    &.gux-arrow-position-left {
      order: -1;
      margin: 0 var(--gux-spacing-xs) 0 0;
    }

    &.gux-expanded {
      transform: rotate(-180deg);
      color: var(--gse-ui-accordion-header-default-foreground-chevronIcon-open);
    }
  }

  &:focus-visible {
    outline: var(--gse-ui-accordion-focusBorder-width)
      var(--gse-ui-accordion-focusBorder-style)
      var(--gse-ui-accordion-focusBorder-color);
    border-radius: var(--gse-ui-accordion-menuItem-focus-borderRadius);
    outline-offset: var(--gse-ui-accordion-focus-offset-gap);
  }

  &:hover {
    .gux-header-icon {
      color: var(
        --gse-ui-accordion-header-default-foreground-chevronIcon-hover
      );
    }
  }
}

.gux-content {
  box-sizing: border-box;
  display: none;
  border-bottom: var(--gse-ui-accordion-wrapper-dividerBorder-width)
    var(--gse-ui-accordion-wrapper-dividerBorder-style)
    var(--gse-ui-accordion-wrapper-dividerBorder-color);

  &.gux-expanded {
    display: block;
  }

  &.gux-text-content-layout {
    padding: var(--gux-spacing-medium);
  }
}
