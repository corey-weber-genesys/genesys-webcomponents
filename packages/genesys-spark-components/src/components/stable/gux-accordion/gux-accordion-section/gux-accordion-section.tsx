import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop,
  State,
  Watch
} from '@stencil/core';

import { randomHTMLId } from '@utils/dom/random-html-id';
import { logError } from '@utils/error/log-error';

import {
  GuxAccordionSectionArrowPosition,
  GuxAccordionSectionContentLayout
} from './gux-accordion-section.types';

/**
 * @slot header - Required slot for the heading
 * @slot subheader - Optional slot for a subheader
 * * @slot icon - Optional slot for an icon
 */

@Component({
  styleUrl: 'gux-accordion-section.scss',
  tag: 'gux-accordion-section',
  shadow: true
})
export class GuxAccordionSection {
  private sectionId: string = randomHTMLId('gux-accordion-section');
  private headerId: string = randomHTMLId('gux-accordion-header');
  private hasIconSlot: boolean;

  @Element()
  root: HTMLElement;

  /**
   * Position of the arrow chevron icon. Position can be 'default' or 'before-text'.  'beside-text' is deprecated and will be removed in v4.
   */
  @Prop()
  arrowPosition: GuxAccordionSectionArrowPosition = 'default';

  /**
   * The content layout used in the accordion section. 'text' layout provides default padding, 'custom' removes default padding.
   */
  @Prop()
  contentLayout: GuxAccordionSectionContentLayout = 'text';

  @Prop({ mutable: true })
  open: boolean = false;

  @Prop()
  disabled: boolean = false;

  @Prop()
  reverseHeadings: boolean = false;

  @Event()
  internalsectionopened: EventEmitter<void>;

  @Watch('open')
  watchOpen(open: boolean): void {
    if (open) {
      this.internalsectionopened.emit();
    }
  }

  @State()
  headingLevel: number;

  private toggle() {
    this.open = !this.open;
  }

  private isArrowPositionBeforeText(): boolean {
    return this.arrowPosition === 'before-text';
  }

  // arrow position 'beside-text' will be removed in v4 (COMUI-1128)
  private isArrowPositionedBesideText(): boolean {
    return this.arrowPosition === 'beside-text';
  }

  private handleSlotChange(slotname: 'header' | 'subheader'): void {
    const slot = this.root.querySelector(`[slot="${slotname}"]`);

    if (!slot || !/^H[1-6]$/.test(slot.nodeName)) {
      logError(
        this.root,
        `For accessibility reasons the ${slotname} slot should be filled with a HTML heading tag (h1 - h6).`
      );
    }

    if (slotname == 'header') {
      this.headingLevel = parseInt(slot.nodeName.replace('H', ''), 10);
    }
  }

  componentWillLoad() {
    this.hasIconSlot = !!this.root.querySelector('[slot="icon"]');
  }

  render(): JSX.Element {
    return (
      <section class={{ 'gux-disabled': this.disabled }}>
        <div id={this.headerId} role="heading" aria-level={this.headingLevel}>
          <button
            class={{
              'gux-header': true,
              'gux-reverse-headings': this.reverseHeadings
            }}
            type="button"
            aria-expanded={this.open.toString()}
            aria-controls={this.sectionId}
            disabled={this.disabled}
            onClick={this.toggle.bind(this)}
          >
            {this.hasIconSlot && <slot name="icon"></slot>}

            <div
              class={{
                'gux-header-text': true,
                'gux-arrow-position-beside': this.isArrowPositionedBesideText()
              }}
            >
              <slot
                onSlotchange={() => this.handleSlotChange('header')}
                name="header"
              ></slot>
              <slot
                onSlotchange={() => this.handleSlotChange('subheader')}
                name="subheader"
              ></slot>
            </div>
            <div
              class={{
                'gux-header-icon': true,
                'gux-expanded': this.open,
                'gux-arrow-position-before-text':
                  this.isArrowPositionBeforeText()
              }}
            >
              <gux-icon decorative icon-name="chevron-small-down"></gux-icon>
            </div>
          </button>
        </div>

        <div
          id={this.sectionId}
          role="region"
          aria-labelledby={this.headerId}
          class={{
            'gux-content': true,
            'gux-expanded': this.open,
            'gux-text-content-layout': this.contentLayout === 'text'
          }}
        >
          <slot name="content"></slot>
        </div>
      </section>
    ) as JSX.Element;
  }
}
