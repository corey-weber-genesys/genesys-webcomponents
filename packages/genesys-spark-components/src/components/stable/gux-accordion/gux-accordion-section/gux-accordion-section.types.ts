export type GuxAccordionSectionArrowPosition = 'start' | 'end';

export type GuxAccordionSectionContentLayout = 'text' | 'custom';
