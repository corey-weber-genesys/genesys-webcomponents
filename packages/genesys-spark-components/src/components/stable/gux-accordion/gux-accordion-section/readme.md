# gux-accordion

<!-- Auto Generated Below -->


## Properties

| Property          | Attribute          | Description                                                                                                                 | Type                 | Default  |
| ----------------- | ------------------ | --------------------------------------------------------------------------------------------------------------------------- | -------------------- | -------- |
| `arrowPosition`   | `arrow-position`   | Position of the arrow chevron icon. Position can be 'start' or 'end'.                                                       | `"end" \| "start"`   | `'end'`  |
| `contentLayout`   | `content-layout`   | The content layout used in the accordion section. 'text' layout provides default padding, 'custom' removes default padding. | `"custom" \| "text"` | `'text'` |
| `disabled`        | `disabled`         |                                                                                                                             | `boolean`            | `false`  |
| `open`            | `open`             |                                                                                                                             | `boolean`            | `false`  |
| `reverseHeadings` | `reverse-headings` |                                                                                                                             | `boolean`            | `false`  |


## Events

| Event                   | Description | Type                |
| ----------------------- | ----------- | ------------------- |
| `internalsectionopened` |             | `CustomEvent<void>` |


## Slots

| Slot          | Description                     |
| ------------- | ------------------------------- |
| `"header"`    | Required slot for the heading   |
| `"icon"`      | Optional slot for an icon       |
| `"subheader"` | Optional slot for a subheader * |


## Dependencies

### Depends on

- [gux-icon](../../gux-icon)

### Graph
```mermaid
graph TD;
  gux-accordion-section --> gux-icon
  style gux-accordion-section fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
