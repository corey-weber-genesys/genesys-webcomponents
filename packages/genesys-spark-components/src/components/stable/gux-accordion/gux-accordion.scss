gux-accordion {
  -custom-noop: noop;
}
