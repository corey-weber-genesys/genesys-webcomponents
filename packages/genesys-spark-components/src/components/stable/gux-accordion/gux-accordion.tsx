import { Component, Element, h, JSX, Listen, Prop } from '@stencil/core';

import { trackComponent } from '@utils/tracking/usage';

/**
 * @slot - collection of gux-accordion-section elements
 */

@Component({
  styleUrl: 'gux-accordion.scss',
  tag: 'gux-accordion',
  shadow: true
})
export class GuxAccordion {
  @Element()
  root: HTMLElement;

  @Prop()
  singleOpenSection: boolean = false;

  @Listen('internalsectionopened')
  handleInternalsectionopened(event: CustomEvent) {
    event.stopImmediatePropagation();

    if (this.singleOpenSection) {
      this.getAccordionSections().forEach(section => {
        if (section !== event.target) {
          this.closeSection(section);
        }
      });
    }
  }

  componentWillLoad() {
    if (this.singleOpenSection) {
      this.getAccordionSections().reduceRight((openFound, section) => {
        if (openFound) {
          this.closeSection(section);
        }

        return openFound || section.open;
      }, false);
    }

    trackComponent(this.root);
  }

  private getAccordionSections(): HTMLGuxAccordionSectionElement[] {
    return Array.from(this.root.children) as HTMLGuxAccordionSectionElement[];
  }

  private closeSection(section: HTMLGuxAccordionSectionElement): void {
    if (!section.disabled) {
      section.open = false;
    }
  }

  render(): JSX.Element {
    return (<slot></slot>) as JSX.Element;
  }
}
