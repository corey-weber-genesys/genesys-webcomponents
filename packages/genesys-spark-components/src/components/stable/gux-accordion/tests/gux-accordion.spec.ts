import { newSpecPage } from '@stencil/core/testing';
import { GuxAccordion } from '../gux-accordion';
import { GuxAccordionSection } from '../gux-accordion-section/gux-accordion-section';

const components = [GuxAccordion, GuxAccordionSection];
const html = `
<gux-accordion>
  <gux-accordion-section>
    <h2 slot="header">First Section</h2>
    <div slot="content">Sample Content</div>
  </gux-accordion-section>
  <gux-accordion-section arrow-position="beside-text">
    <h2 slot="header">Second Section</h2>
    <div slot="content">Sample Content</div>
  </gux-accordion-section>
  <gux-accordion-section disabled>
    <h2 slot="header">Third Section</h2>
    <h3 slot="subheader">This is the subheader</h3>
    <div slot="content">Sample Content</div>
  </gux-accordion-section>
  <gux-accordion-section>
    <h2 slot="header">Fourth Section</h2>
    <h3 slot="subheader">This is the subheader</h3>
    <gux-icon
      slot="icon"
      icon-name="user-add"
      screenreader-text="add John Smith to contact list"
    ></gux-icon>
    <div slot="content">Sample Content</div>
  </gux-accordion-section>
  <gux-accordion-section reverse-headings="true">
    <h2 slot="header">Fourth Section</h2>
    <h3 slot="subheader">This is the subheader</h3>
    <gux-icon
      slot="icon"
      icon-name="user-add"
      screenreader-text="add John Smith to contact list"
    ></gux-icon>
    <div slot="content">Sample Content</div>
  </gux-accordion-section>
</gux-accordion>
`;
const language = 'en';

describe('gux-accordion', () => {
  it('should build', async () => {
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxAccordion);
  });

  it('renders', async () => {
    const page = await newSpecPage({ components, html, language });

    expect(page.root).toMatchSnapshot();
  });
});
