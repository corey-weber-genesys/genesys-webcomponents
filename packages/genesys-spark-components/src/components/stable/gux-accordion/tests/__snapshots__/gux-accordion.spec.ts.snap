// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-accordion renders 1`] = `
<gux-accordion>
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <gux-accordion-section>
    <mock:shadow-root>
      <section>
        <div id="gux-accordion-header-i" role="heading">
          <button aria-controls="gux-accordion-section-i" aria-expanded="false" class="gux-header" type="button">
            <div class="gux-header-text">
              <slot name="header"></slot>
              <slot name="subheader"></slot>
            </div>
            <div class="gux-header-icon">
              <gux-icon decorative="" icon-name="chevron-small-down"></gux-icon>
            </div>
          </button>
        </div>
        <div aria-labelledby="gux-accordion-header-i" class="gux-content gux-text-content-layout" id="gux-accordion-section-i" role="region">
          <slot name="content"></slot>
        </div>
      </section>
    </mock:shadow-root>
    <h2 slot="header">
      First Section
    </h2>
    <div slot="content">
      Sample Content
    </div>
  </gux-accordion-section>
  <gux-accordion-section arrow-position="beside-text">
    <mock:shadow-root>
      <section>
        <div id="gux-accordion-header-i" role="heading">
          <button aria-controls="gux-accordion-section-i" aria-expanded="false" class="gux-header" type="button">
            <div class="gux-arrow-position-beside gux-header-text">
              <slot name="header"></slot>
              <slot name="subheader"></slot>
            </div>
            <div class="gux-header-icon">
              <gux-icon decorative="" icon-name="chevron-small-down"></gux-icon>
            </div>
          </button>
        </div>
        <div aria-labelledby="gux-accordion-header-i" class="gux-content gux-text-content-layout" id="gux-accordion-section-i" role="region">
          <slot name="content"></slot>
        </div>
      </section>
    </mock:shadow-root>
    <h2 slot="header">
      Second Section
    </h2>
    <div slot="content">
      Sample Content
    </div>
  </gux-accordion-section>
  <gux-accordion-section disabled="">
    <mock:shadow-root>
      <section class="gux-disabled">
        <div id="gux-accordion-header-i" role="heading">
          <button aria-controls="gux-accordion-section-i" aria-expanded="false" class="gux-header" disabled="" type="button">
            <div class="gux-header-text">
              <slot name="header"></slot>
              <slot name="subheader"></slot>
            </div>
            <div class="gux-header-icon">
              <gux-icon decorative="" icon-name="chevron-small-down"></gux-icon>
            </div>
          </button>
        </div>
        <div aria-labelledby="gux-accordion-header-i" class="gux-content gux-text-content-layout" id="gux-accordion-section-i" role="region">
          <slot name="content"></slot>
        </div>
      </section>
    </mock:shadow-root>
    <h2 slot="header">
      Third Section
    </h2>
    <h3 slot="subheader">
      This is the subheader
    </h3>
    <div slot="content">
      Sample Content
    </div>
  </gux-accordion-section>
  <gux-accordion-section>
    <mock:shadow-root>
      <section>
        <div id="gux-accordion-header-i" role="heading">
          <button aria-controls="gux-accordion-section-i" aria-expanded="false" class="gux-header" type="button">
            <slot name="icon"></slot>
            <div class="gux-header-text">
              <slot name="header"></slot>
              <slot name="subheader"></slot>
            </div>
            <div class="gux-header-icon">
              <gux-icon decorative="" icon-name="chevron-small-down"></gux-icon>
            </div>
          </button>
        </div>
        <div aria-labelledby="gux-accordion-header-i" class="gux-content gux-text-content-layout" id="gux-accordion-section-i" role="region">
          <slot name="content"></slot>
        </div>
      </section>
    </mock:shadow-root>
    <h2 slot="header">
      Fourth Section
    </h2>
    <h3 slot="subheader">
      This is the subheader
    </h3>
    <gux-icon icon-name="user-add" screenreader-text="add John Smith to contact list" slot="icon"></gux-icon>
    <div slot="content">
      Sample Content
    </div>
  </gux-accordion-section>
  <gux-accordion-section reverse-headings="true">
    <mock:shadow-root>
      <section>
        <div id="gux-accordion-header-i" role="heading">
          <button aria-controls="gux-accordion-section-i" aria-expanded="false" class="gux-header gux-reverse-headings" type="button">
            <slot name="icon"></slot>
            <div class="gux-header-text">
              <slot name="header"></slot>
              <slot name="subheader"></slot>
            </div>
            <div class="gux-header-icon">
              <gux-icon decorative="" icon-name="chevron-small-down"></gux-icon>
            </div>
          </button>
        </div>
        <div aria-labelledby="gux-accordion-header-i" class="gux-content gux-text-content-layout" id="gux-accordion-section-i" role="region">
          <slot name="content"></slot>
        </div>
      </section>
    </mock:shadow-root>
    <h2 slot="header">
      Fourth Section
    </h2>
    <h3 slot="subheader">
      This is the subheader
    </h3>
    <gux-icon icon-name="user-add" screenreader-text="add John Smith to contact list" slot="icon"></gux-icon>
    <div slot="content">
      Sample Content
    </div>
  </gux-accordion-section>
</gux-accordion>
`;
